src/target/riscv/riscv.c

```diff
@@ -534,9 +534,12 @@ static int jim_configure_ebreak(struct riscv_private_config *config, struct jim_
 		/* Here a common "ebreak" action is processed, e.g:
 		 * "riscv.cpu configure -ebreak halt"
 		 */
+		int res = jim_getopt_obj(goi, NULL);
+		if (res != JIM_OK)
+			return res;
 		for (int ebreak_ctl_i = 0; ebreak_ctl_i < N_RISCV_MODE; ++ebreak_ctl_i)
 			config->dcsr_ebreak_fields[ebreak_ctl_i] = common_mode_nvp->value;
-		return jim_getopt_obj(goi, NULL);
+		return JIM_OK;
 	}
```

Comment:
(No change requested - just a question for my understanding)

Do I understand correctly that `jim_getopt_obj(goi, ptr_to_Jim_Obj_or_NULL)` is supposed to pop the first argument in the argument list (goi), and put it into the Jim_Obj pointed by the second argument (if not NULL)?

Does it mean that if we wanted to strictly obey the JimTcl API, we would not be reading `goi->argv` directly, but instead we would get the values of arguments using jim_getopt_obj(), as done e.g. [here](https://github.com/riscv-collab/riscv-openocd/blob/41a225460c3b9a6c1f61a0777f101ff009f56007/src/target/arm_tpiu_swo.c#L971-L977)?

Reply:
Ok, thanks for explanation. I don't mind the direct access to `goi->argv`, I was just curious.